tests/tests/metrics.rs

```diff
@@ -60,30 +62,38 @@ pub struct CounterMiddleware<S> {
 #[async_trait]
 impl<'a, S> RpcServiceT<'a> for CounterMiddleware<S>
 where
-	S: RpcServiceT<'a> + Send + Sync,
+	S: RpcServiceT<'a> + Send + Sync + Clone + 'static,
```

Comment:
//cc @jsdw do you know if it's possible to avoid cloning the service to move it to the async block below?

I'm probably need a lifetime on the future for that to work here and bummer that it worked with async_trait :thinking:

Reply:
As it stands, with the service being taken as a reference through `&self`, I can't see how you'd avoid cloning it unless you put it into a `RefCell`/`Mutex` and moved it out into the `async` block and then put it back at the end (but of course, since `call` only takes `&self`, there's nothing that prevents it being called more than once at a time at compile time at least, which would lead to issues!)

Reply:
It should be quite cheap to clone the underlying service RpcService anyway, it's cloning a couple of Arc's but if someone puts a bunch of heavy stuff in a wrapped service it could be expensive...

Reply:
At least we give them the choice; they could choose to put their service behind an `Arc` of could avoid it entirely if it's a cheap service that is trivially cloneable :)